appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/model/ExtensibleImpl.java

```diff
@@ -42,50 +42,81 @@
 import static fish.payara.microprofile.openapi.impl.model.util.ModelUtils.isAnnotationNull;
 import static fish.payara.microprofile.openapi.impl.model.util.ModelUtils.mergeProperty;
 
-import java.util.HashMap;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
 import java.util.LinkedHashMap;
 import java.util.Map;
 
 import org.eclipse.microprofile.openapi.annotations.extensions.Extension;
 import org.eclipse.microprofile.openapi.models.Extensible;
 
-public abstract class ExtensibleImpl implements Extensible {
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.databind.ObjectMapper;
 
-    protected Map<String, Object> extensions = new HashMap<>();
+public abstract class ExtensibleImpl<T extends Extensible<T>> implements Extensible<T> {
 
+    protected Map<String, Object> extensions = new LinkedHashMap<>();
+
+    @JsonAnyGetter
     @Override
     public Map<String, Object> getExtensions() {
         return extensions;
     }
 
+    @SuppressWarnings("unchecked")
+    @Override
+    public T addExtension(String name, Object value) {
+        if (value != null) {
+            extensions.put(name, value);
+        }
+        return (T) this;
+    }
+
     @Override
-    public void addExtension(String name, Object value) {
-        extensions.put(name, value);
+    public void removeExtension(String name) {
+        extensions.remove(name);
     }
 
     @Override
     public void setExtensions(Map<String, Object> extensions) {
         this.extensions = extensions;
     }
 
-    public static void merge(Extension from, Extensible to, boolean override) {
+    public static void merge(Extension from, Extensible<?> to, boolean override) {
         if (isAnnotationNull(from)) {
             return;
         }
         if (to.getExtensions() == null) {
             to.setExtensions(new LinkedHashMap<>());
         }
         if (from.name() != null && !from.name().isEmpty()) {
-            Object value = mergeProperty(to.getExtensions().get(from.name()), convertExtensionValue(from.value()),
-                    override);
+            Object value = mergeProperty(to.getExtensions().get(from.name()), 
+                    convertExtensionValue(from.value(), from.parseValue()), override);
             to.getExtensions().put(from.name(), value);
         }
     }
 
-    public static Object convertExtensionValue(String value) {
+    public static Object convertExtensionValue(String value, boolean parseValue) {
         if (value == null) {
             return null;
         }
+        if (parseValue) {
```

Comment:
New feature in 1.1 where value should be parsed to some other type than `String`.

Reply:
I'm afraid it is much more complicated than that, the spec says [any valid JSON value](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#specificationExtensions), which entails for exable:

* leading and trailing spaces on lists and objects
* strings containing dots while not being numbers
* NaN being a double

I think the best approach is to always pass it to Jackson, and then decide the return value on the type of JsonNode.

If it fails, it is a string

Reply:
Turn out one can't rely on the API javadoc which listed the 4 cases I was handling. 
Good suggestion. Will change it that way.